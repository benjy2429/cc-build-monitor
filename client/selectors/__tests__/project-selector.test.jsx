import React from 'react';
import { shallow } from 'enzyme';
import projectSelector from '../project-selector';

const WrappedComponent = () => <div />;
const ProjectSelector = projectSelector(WrappedComponent);

describe('Project Selector', () => {
  it('renders the wrapped component', () => {
    const component = shallow(
      <ProjectSelector projects={[]} />,
    );
    expect(component).toMatchSnapshot();
  });

  it('sorts projects by activity', () => {
    const projects = [
      { activity: 'Sleeping' },
      { activity: 'Building' },
      { activity: 'SomethingElse' },
    ];
    const component = shallow(
      <ProjectSelector projects={projects} />,
    );
    expect(component.find(WrappedComponent).prop('projects')).toMatchSnapshot();
  });

  it('sorts projects by build status', () => {
    const projects = [
      { buildStatus: 'SomethingElse' },
      { buildStatus: 'Success' },
      { buildStatus: 'Failure' },
      { buildStatus: 'Unknown' },
    ];
    const component = shallow(
      <ProjectSelector projects={projects} />,
    );
    expect(component.find(WrappedComponent).prop('projects')).toMatchSnapshot();
  });
});
