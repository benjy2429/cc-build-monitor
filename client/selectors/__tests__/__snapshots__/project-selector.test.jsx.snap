// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Selector renders the wrapped component 1`] = `
<WrappedComponent
  projects={
    Object {
      "building": Array [],
      "failing": Array [],
      "passing": Array [],
      "unknown": Array [],
    }
  }
/>
`;

exports[`Project Selector sorts projects by activity 1`] = `
Object {
  "building": Array [
    Object {
      "activity": "Building",
    },
  ],
  "failing": Array [],
  "passing": Array [],
  "unknown": Array [
    Object {
      "activity": "Sleeping",
    },
    Object {
      "activity": "SomethingElse",
    },
  ],
}
`;

exports[`Project Selector sorts projects by build status 1`] = `
Object {
  "building": Array [],
  "failing": Array [
    Object {
      "lastBuildStatus": "Failure",
    },
  ],
  "passing": Array [
    Object {
      "lastBuildStatus": "Success",
    },
  ],
  "unknown": Array [
    Object {
      "lastBuildStatus": "SomethingElse",
    },
    Object {
      "lastBuildStatus": "Unknown",
    },
  ],
}
`;
